### PR TITLE
docs: doc/promtail/journal: adds available labels paragraph

### LINKED ISSUE #2048: Please document ____journal__* labels
**Is your feature request related to a problem? Please describe.**

I'm missing documentation of which label I can get from Promtail's journal scraping. Searching the repo I found __journal__systemd_unit and __journal__hostname, but are there more ? 

**Describe the solution you'd like**

Update https://github.com/grafana/loki/blob/master/docs/clients/promtail/configuration.md#journal_config with the same kind of listing there's around https://github.com/grafana/loki/blob/master/docs/clients/promtail/configuration.md#node

### PR DESCRIPTION
**Pull request by arcenik**
* Adds journal available labels paragraph

**What this PR does / why we need it**:

Adds a paragraph promtail/journal documentation about __journal_* labels.

**Which issue(s) this PR fixes**:
Fixes #2048 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [x] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [x] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/send-data/promtail/configuration.md [hunk 1]
```diff
@@ -853,14 +853,51 @@ using the AMD64 Docker image, this is enabled by default.
...
-Priority label is available as both value and keyword. For example, if `priority` is `3` then the labels will be `__journal_priority` with a value `3` and `__journal_priority_keyword` with a corresponding keyword `err`.
-{{% /admonition %}}
+Labels are imported from systemd-journal fields. The label name is the field name set to lower case with **__journal_** prefix. See the man page [systemd.journal-fields](https://www.freedesktop.org/software/systemd/man/latest/systemd.journal-fields.html) for more information.
+
+For example:
+
+| journal field      | label                   |
+|--------------------|-------------------------|
+| _HOSTNAME          | __journal__hostname     |
+| _SYSTEMD_USER_UNIT | __journal__systemd_unit |
```
Comment (JStickler) [inline docs/sources/send-data/promtail/configuration.md:886]: Just double checking, the label here is `__journal__systemd_unit` and not `__journal__systemd_user_unit`?

File: docs/sources/send-data/promtail/configuration.md [hunk 2]
```diff
@@ -853,14 +853,51 @@ using the AMD64 Docker image, this is enabled by default.
...
+| ERRNO              | __journal_errno         |
+
+In addition to **__journal_priority** (imported from **PRIORITY** journal field, where the value is an integer from 0 to 7), promtail adds **__journal_priority_keyword** label where the value is generated using the **makeJournalPriority** mapping function.
```
Comment (JStickler) [inline docs/sources/send-data/promtail/configuration.md:889]: https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#text-formatting 
```suggestion
In addition to `__journal_priority` (imported from `PRIORITY` journal field, where the value is an integer from 0 to 7), promtail adds `__journal_priority_keyword` label where the value is generated using the `makeJournalPriority` mapping function.
```

File: docs/sources/send-data/promtail/configuration.md (hunk 1 above)
Comment (arcenik) [reply docs/sources/send-data/promtail/configuration.md:886]: indeed

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Pull request merged**